### PR TITLE
Rename server pkg to contextx

### PR DESCRIPTION
**Pull request by gonzaloserrano**
Also get rid of the `daemon` which AFAIK was not used anyway and also it's not a daemon anyway.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,11 +6,11 @@ There are two types of packages:
...
-| Package            | Type  | Description                                                        |
-| ---                | ----  | -----------                                                        |
-| [`httpx`](httpx)   | `ext` | Extends package `net/http` with routing utilities, decorators, etc |
-| [`logx`](logx)     | `ext` | Adds structured logging and logstash support to `log`              |
-| [`server`](server) | `new` | Adds types like `Runner` to run application servers                |
+| Package            | Type  | Description                                                         |
+| ---                | ----  | -----------                                                         |
+| [`httpx`](httpx)   | `ext` | Extends package `net/http` with routing utilities, decorators, etc  |
+| [`logx`](logx)     | `ext` | Adds structured logging and logstash support to `log`               |
+| [`runner`](runner) | `new` | Adds types like `Runner` and to run stuff given a `context.Context` |
```
Comment (ronnylt) [inline README.md]: Typo in the description

File: runner/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Package server
+# Package runner
 
-`package server` provides the types needed to run application servers.
+`package runner` provides the types needed to run stuff.
```
Comment (ronnylt) [inline runner/README.md]: I would extend this description with more details, i.e. explaining why it's important to run stuff this way

Comment (gonzaloserrano) [inline runner/README.md]: Any help with this is welcome!

**Comment by gonzaloserrano** [comment]
Improved godoc as suggested.

**Comment by rogerclotet** [comment]
The ticker file is using the old "server": https://github.com/gonzaloserrano/bsk/pull/1

**Comment by gonzaloserrano** [comment]
Ready to merge, thanks @rogerclotet for the fix.

**Comment by rogerclotet** [comment]
golint is still failing: "runner/runner.go:15:6: type name will be used as runner.RunnerFunc by other packages, and that stutters; consider calling this Func". I don't know if the standard in those cases would be to name it Func, it seems weird being an uppercase version of a reserved word.

**Comment by gonzaloserrano** [comment]
Ouch! Yeah you told me about this before. What do you think about naming it `RunFunc`, even is not perfect?

**Comment by rogerclotet** [comment]
I still would prefer Func even if it feels a bit weird, because RunFunc seems to indicate a different thing. Without looking at the implementation I would say it takes a function and runs it, or something like that.

**Comment by gonzaloserrano** [comment]
What if we change the pkg name to `run` ?

**Comment by sp-ludovic-ivain** [comment]
``` `package runner` provides abstractions needed to run tasks```

what about ```runner.Task(...)``` ?

**Comment by gonzaloserrano** [comment]
I've renamed it to `run`, what do you think?

p.s sorry @sp-ludovic-ivain I didn't see your comment before the change
p.p.s it's better that you use your non-sp user ;)

**Comment by ronnylt** [comment]
@socialpoint-labs/collaborators Do you think  the abstractions provided by this package are still needed? 

- Posix signals adapter -> Could go in a `signalx` package 
- `Runner` abstraction -> Now I see it as an extension of the stdlib's `context` package. What about adding a ' contextx' package ?

**Comment by gonzaloserrano** [comment]
I really like to rename it to `contextx`, makes sense!

Not sure about moving the decorators to their own packages (for now signals) that depend on `contextx`. I think for now I would leave it in the same package.

**Comment by gonzaloserrano** [comment]
Renamed to `contextx`

File: contextx/runner.go [hunk 3]
```diff
@@ -20,18 +20,16 @@ func (f RunnerFunc) Run(ctx context.Context) {
 	f(ctx)
 }
 
-// EmptyRunner is a Runner that does nothing for testing purposes only.
-func EmptyRunner() RunnerFunc {
+// Empty is a Runner that does nothing for testing purposes only.
+func Empty() RunnerFunc {
```
Comment (neomede) [inline contextx/runner.go]: contextx.Empty() looks weird when you use it, I prefer contextx.EmptyRunner() (same for MultiRunner)

**Pull request merged**